.github/workflows/lint.yml

```
@@ -0,0 +1,11 @@
+name: Lint
+on: [pull_request]
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Markdown Lint
+      uses: nosborn/github-action-markdown-cli@v1.1.1
```

I may be a little paranoid, but I like to pin actions by SHA, as [recommended by the GHA docs](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsuses):
```suggestion
      uses: nosborn/github-action-markdown-cli@9fc95163471d6460c35cccad13645912aaa25d5f
```

Note also that:
i) 1.1.1 is shown as "Unverified" on GitHub:
> This commit is not signed, but one or more authors requires that any commit attributed to them is signed.

ii) 2.0.0 of the action has been released, perhaps this commit should add a dependabot config for the GHA?
That's as easy as adding a `.github/dependabot.yml` with contents like:
```
version: 2
updates:
- package-ecosystem: github-actions
  directory: "/"
  schedule:
    interval: daily
```

---

cc @jeffmendoza and @laurentsimon who have been working on some automation to help enforce/automate the GitHub actions pinning stuff.

---

@joshuagl suggestion is the one we recommend in scorecard too, e.g. https://github.com/ossf/scorecard/blob/main/.github/workflows/codeql-analysis.yml#L52 just at end, we also do "  # version x.y.z" for reference as otherwise hashes are not human readable.

---

Thanks! Done. (Note that the Dockerfile still fetches things without a specific version tag (jq and curl) but it's better than nothing.)

---

More important than pinning, is restricting permissions with something like
```
permissions:
  contents: read
```
https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#permissions
https://docs.github.com/en/github/administering-a-repository/managing-repository-settings/disabling-or-limiting-github-actions-for-a-repository#setting-the-permissions-of-the-github_token-for-your-repository

---

Thanks for the links!